src/services/queue-service.js

```diff
@@ -1,10 +1,40 @@
 import { UserNetworkInstance } from './network';
+import { setKeyWithExpiryToLocalStore } from '../utilities/local-storage';
+import { log } from "../utilities/log";
 
+//const DEFAULT_POLLING_DELAY_IN_SECONDS = 600;
+const DEFAULT_POLLING_DELAY_IN_SECONDS = 10;
+var currentPollingDelayInSeconds = DEFAULT_POLLING_DELAY_IN_SECONDS;
+var checkInProgress = false;
+
+export const userQueueNextPullCheckLocalStoreName = "gist.web.userQueueNextPullCheck";
 export async function getUserQueue() {
+  var response;
   try {
-    var response = await UserNetworkInstance().post(`/api/v1/users`, {});
-    return response;
+    if (!checkInProgress) {
+      checkInProgress = true;
+      response = await UserNetworkInstance().post(`/api/v1/users`, {});
+    }
   } catch (error) {
-    return error.response;
+    if (error.response) {
+      response = error.response;
+    } else {
+      log(`Error getting user queue: ${error}`);
+    }
+  } finally {
+    checkInProgress = false;
+    scheduleNextQueuePull(response);
+    return response;
+  }
+}
+
+function scheduleNextQueuePull(response) {
+  if (response && response.headers) {
+    var pollingInterval = response.headers['x-gist-queue-polling-interval'];
+    if (pollingInterval && pollingInterval > 0) {
+      currentPollingDelayInSeconds = pollingInterval;
+    }
   }
+  var expiryDate = new Date(new Date().getTime() + currentPollingDelayInSeconds * 1000);
+  setKeyWithExpiryToLocalStore(userQueueNextPullCheckLocalStoreName, currentPollingDelayInSeconds, expiryDate);
```

Comment:
Out of scope of this PR, but we should look in Cross tab synchronization.

or rely on something like https://github.com/nanostores/persistent

Reply:
Isn't that the default behavior? I just checked on Chrome and both tabs were showing the same values in local store.

Reply:
Yep I was thinking of updating other pollers with the latest change on a [StorageEvent](https://developer.mozilla.org/en-US/docs/Web/API/StorageEvent). 

But just realized we're checking localstorage every second. Overkill 🤦🏽 

Thanks.